Yank/repex.py

```diff
@@ -1181,7 +1237,9 @@ class while the simulation is running. This reads the SamplerStates of every
     # Constructors.
     # -------------------------------------------------------------------------
 
-    def __init__(self, mcmc_moves=None, number_of_iterations=1, replica_mixing_scheme='swap-all'):
+    def __init__(self, mcmc_moves=None, number_of_iterations=1, replica_mixing_scheme='swap-all',
+                 online_analysis_interval=None, online_analysis_target_error=1.0,
```

Comment:
> I'm still wanting to put tests in, but I cant think of how to easily do that since this relies on the analyze module which we have no tests for anyways because they will all be slow and require real simulation data.

What about taking a small system in implicit solvent or vacuum and pre-generating some data for it that gets checked into the repo? You can test (1) offline analysis, and (2) online analysis (to make sure it terminates quickly if the simulation has already converged to the specified tolerance).